CMakeLists.txt

```diff
@@ -60,6 +61,62 @@ option(ENABLE_CUFFT_BACKEND "Enable the cuFFT backend for the DFT interface" OFF
 option(ENABLE_ROCFFT_BACKEND "Enable the rocFFT backend for the DFT interface" OFF)
 option(ENABLE_PORTFFT_BACKEND "Enable the portFFT DFT backend for the DFT interface. Cannot be used with other DFT backends." OFF)
 
+# Following if-conditions allow to decouple cmake configuration variables with
+# the corresponding generated macro. This is done to be conformant with ES.33
+# C++ Core Guidelines
+if(ENABLE_MKLCPU_BACKEND)
+  option(ENABLE_MKLCPU_THREAD_TBB "Enable the use of Intel TBB with the oneMKL CPU backend" ON)
+  set(ONEAPI_ONEMKL_ENABLE_MKLCPU_BACKEND True)
+endif()
+
+if(ENABLE_MKLGPU_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_MKLGPU_BACKEND True)
+endif()
+
+if(ENABLE_CUBLAS_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_CUBLAS_BACKEND True)
+endif()
+
+if(ENABLE_ROCBLAS_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_ROCBLAS_BACKEND True)
+endif()
+
+if(ENABLE_NETLIB_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_NETLIB_BACKEND True)
+endif()
+
+if(ENABLE_PORTBLAS_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_PORTBLAS_BACKEND True)
+endif()
+
+if(ENABLE_CURAND_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_CURAND_BACKEND True)
+endif()
+
+if(ENABLE_ROCRAND_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_ROCRAND_BACKEND True)
+endif()
+
+if(ENABLE_CUSOLVER_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_CUSOLVER_BACKEND True)
+endif()
+
+if(ENABLE_ROCSOLVER_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_ROCSOLVER_BACKEND True)
+endif()
+
+if(ENABLE_CUFFT_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_CUFFT_BACKEND True)
+endif()
+
+if(ENABLE_ROCFFT_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_ROCFFT_BACKEND True)
+endif()
+
+if(ENABLE_PORTFFT_BACKEND)
+  set(ONEAPI_ONEMKL_ENABLE_PORTFFT_BACKEND True)
+endif()
```

Comment:
I think we're missing the defines for `ONEAPI_ONEMKL_REF_BLAS_LIBNAME` and `ONEAPI_ONEMKL_REF_CBLAS_LIBNAME` here.